### DISCUSSION TITLE
Spotlight mode not saved across sessions?

### DISCUSSION THREAD

**Discussion by autoca** [freeplane/freeplane#626] [Help] [answered]
Hi all,

I noticed that the spotlight mode (View > View settings > Spotlight) does not get remembered across sessions or saved in the map (it seems to be a per-map-setting).

Am I doing something wrong or is this just how it is?
If that's how it is, can I use scripting to have it activate automatically whenever I open a map? If yes, how?
MTIA

Regards, autoca

**Comment by macmarrum** [comment, marked as answer]
It's an application-wide setting: `Preferences…->Appearance->Activate spotlight by default`

_BTW, you can search for commands and/or preferences in Freeplane `Help->Command search`_

If you'd rather have spotlight for some maps but not for others, you can try my experimental library: https://github.com/macmarrum/freeplane-scripting/blob/main/src/ScriptOnMapOpen.groovy

1. Turn off `Preferences…->Appearance->Activate spotlight by default`
2. Download [ScriptOnMapOpen.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/ScriptOnMapOpen.groovy) and place it in `<user-dir>/lib`
3. [Enable script execution](https://docs.freeplane.org/#/scripting/Scripting?id=preparation), if not enabled already
4. Restart Freeplane
5. In your map, on the root node, add an attribute `scriptOnMapOpen`
6. Use `Tools->Edit script…` to enter the script content:
```groovy
menuUtils.executeMenuItems(['SetBooleanMapViewPropertyAction.spotlight'])
```
7. `Actions->Save and exit` the script editor
8. Save and close the map
9. Open it again

**Reply by macmarrum** [reply]
UPDATE: with a missed step (№5)

1. Turn off `Preferences…->Appearance->Activate spotlight by default`
2. Download [ScriptOnMapOpen.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/ScriptOnMapOpen.groovy) and place it in `<user-dir>/lib`
3. [Enable script execution](https://docs.freeplane.org/#/scripting/Scripting?id=preparation), if not enabled already
4. Restart Freeplane
5. In your map, on the root node, add a Node Conditional Style with a Script Filter:
```groovy
ScriptOnMapOpen.nodeConditionalStyle(node)
```
6. In your map, on the root node, add an attribute `scriptOnMapOpen`
7. Use `Tools->Edit script…` to enter the script content:
```groovy
menuUtils.executeMenuItems(['SetBooleanMapViewPropertyAction.spotlight'])
```
8. `Actions->Save and exit` the script editor
9. Save and close the map
10. Open it again

**Comment by autoca** [comment]
Awesome!!!  Thank you so much!!!
